demos/api/core_web_vitals/main.js

```diff
@@ -0,0 +1,182 @@
+  return;
+}
+
+// Set optional form factor.
+if (argv.formFactor) {
+  request.formFactor = argv.formFactor;
+}
+
+/**
+ * Query the CrUX API for metric records.
+ *
+ * @param {{origin?: string, url?: string, formFactor?: string,  metrics?: Array<string>}} requestBody
+ * return {{record: {key: Object, metrics: Map<string, {histogram: Array<{ start: number, end?: number, density: number}>}>}, urlNormalizationDetails: Object}} */
+CrUXApiUtil.query = async function (requestBody) {
+  const endpointUrl = 'https://chromeuxreport.googleapis.com/v1/records:queryRecord';
+  const resp = await fetch(`${endpointUrl}?key=${CrUXApiUtil.KEY}`, {
+    method: 'POST',
+    body: JSON.stringify(requestBody),
+  });
+
+  const json = await resp.json();
+  if (!resp.ok) {
+    console.log(chalk.red(json.error.message));
+    if (resp.status === 404) {
+      // Extra Not Found flavor text.
+      console.log('Data cannot be found for this URL. ' +
+        'Check that the URL you\'re looking up is correct, ' +
+        'don\'t forget "www" if you need it!');
+    }
+    process.exit(1);
+  }
+  return json;
+};
+
+/**
+ * Utility method to transform the response's metric data into an array of usable metric objects
+ * Example return value:
+  [
+    {
+      "acronym": "FCP",
+      "name": "first_contentful_paint",
+      "labeledBins": [{
+        "label": "good",
+        "percentage": 43.51,
+        "start": 0,
+        "end": 1000,
+        "density": 0.4351
+      }, // ... (other bins) ]
+      "percentiles": {
+          "p75": 1792
+      },
+    }, // ... (other metrics) ]
+*
+* @return {{acronym: string, name: string, labelsBins: Array<{label: 'good'|'needs improvement'|'poor', percentage: number, start: number, end?: number, density: number}>}}*/
+function labelMetricData(metrics) {
+  const nameToAcronymMap = {
+    first_contentful_paint: 'FCP',
+    largest_contentful_paint: 'LCP',
+    first_input_delay: 'FID',
+    cumulative_layout_shift: 'CLS',
+  };
+
+  return Object.entries(metrics).map(([metricName, metricData]) => {
+    const standardBinLabels = ['good', 'needs improvement', 'poor'];
+    const metricBins = metricData.histogram;
+
+    // We assume there are 3 histogram bins and they're in order of: good => poor
+    const labeledBins = metricBins.map((bin, i) => {
+      // Assign a good/poor label, calculate a percentage, and add retain all existing bin properties
+      return {
+        label: standardBinLabels[i],
+        percentage: bin.density * 100,
+        ...bin,
+      };
+    });
+
+    return {
+      acronym: nameToAcronymMap[metricName],
+      name: metricName,
+      labeledBins,
+      percentiles: {
+          p75: metricData.percentiles.p75,
+      },
+    };
+  });
+}
+
+// Gather the data and display it.
+(async function () {
+  const json = await CrUXApiUtil.query(request);
+
+  const labeledMetrics = await labelMetricData(json.record.metrics);
+
+  // The CWV 75%ile thresholds.
+  const nameToCWVMap = {
```

Comment:
tbh I'm not sure. We _can_ look at the end of the first bucket of each metric. Which is the CWV threshold. But that isn't guaranteed to be true for all time.

Reply:
related https://github.com/GoogleChrome/web-vitals/issues/2

Reply:
> But that isn't guaranteed to be true for all time.

nothing's a true guarantee, but at this point i can't imagine this would change.

eg the primary cwv image relies on this fact: 
![image](https://user-images.githubusercontent.com/39191/84086225-32c53a00-a99c-11ea-891d-4109a96148d0.png)

Reply:
At minimum, the threshold should not change before May 2021, and if it were to change (in 2021 or some other future CWV release) there'd be a coordinated effort to ensure all tools were updated accordingly.